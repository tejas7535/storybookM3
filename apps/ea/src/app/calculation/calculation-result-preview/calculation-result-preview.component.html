<ng-container *transloco="let t; read: 'calculation'">
  @if (isStandalone$ | ngrxPush) {
    <div class="blur-backdrop" [ngClass]="{ 'scale-y-0': !sticky }"></div>
  }
  <div
    class="left-0 flex w-full items-center justify-center box-border bottom-0 md:bottom-9 z-20 transition-all duration-150"
    [ngClass]="{
      'md:inset-y-auto relative md:fixed max-w-[100vw]':
        (isStandalone$ | ngrxPush) && !sticky,
      'relative h-auto': !isStandalone$ | ngrxPush,
      fixed: isStandalone$ | ngrxPush,
      'preview-mobile': isMobile
    }"
  >
    <div
      *ngrxLet="{
        errors: calculationErrors$ | ngrxPush,
        isCalculationImpossible: isCalculationImpossible$ | ngrxPush,
        isCalculationGeneralError: calculationGeneralError$ | ngrxPush
      } as calculationFailureData"
      class="relative bar bg-background-dark flex flex-col items-center md:flex-row pb-2 px-6 pt-4 md:py-6 md:px-6 rounded-t-lg min-w-full md:min-w-[800px] md:max-w-[calc(100vw-350px)] min-[1000px]:min-w-[890px] drop-shadow-lg z-50 space-y-1 md:space-y-0 overflow-hidden md:rounded-b-lg overflow-y-visible md:mx-4"
      [ngClass]="{
        'rounded-b-lg rounded-none drop-shadow-none md:rounded-lg md:drop-shadow-lg':
          'sticky',
        'mx-auto': !isStandalone$ | ngrxPush
      }"
    >
      <div class="user-hints flex items-center flex-grow-0 flex-shrink-0">
        @if (isCalculationMissingInput$ | ngrxPush) {
          <div class="flex max-w-[180px] flex-row items-center justify-center">
            <div class="flex flex-shrink-0">
              <mat-icon color="primary">info_outline</mat-icon>
            </div>
            <div
              class="text-caption text-high-emphasis ml-2 text-center md:text-left"
            >
              {{ t('calculationMissingInput') }}
            </div>
          </div>
          <mat-divider class="hidden md:block" [vertical]="true"></mat-divider>
        }

        <!-- Calculation Impossible -->
        @if (
          calculationFailureData.isCalculationImpossible ||
          calculationFailureData.isCalculationGeneralError
        ) {
          <div class="flex flex-shrink-0 flex-row items-center">
            @if (calculationFailureData?.errors?.length === 0) {
              <div class="flex flex-shrink-0">
                <mat-icon color="error">warning</mat-icon>
              </div>
            }
            <div
              class="text-caption text-high-emphasis ml-2 flex-shrink-0"
              [style]="{ 'max-width': '220px' }"
            >
              @if (calculationFailureData?.errors?.length === 0) {
                <span> {{ t('calculationImpossible') }} </span>
              }
              @if (calculationFailureData.isCalculationGeneralError) {
                <span> {{ t('generalError') }} </span>
              }
            </div>
          </div>
          <mat-divider [vertical]="true"></mat-divider>
        }
      </div>
      <ng-container>
        @if (overlayData$ | ngrxPush; as overlayData) {
          <div
            class="result-data"
            class="flex touch-pan-x snap-x snap-mandatory flex-row gap-2 md:max-w-auto mx-2 overflow-x-auto max-w-full md:ml-0 md:mr-4"
          >
            @if (calculationFailureData?.errors?.length > 0) {
              <div class="snap-center my-1 md:my-0 min-w-max">
                <ea-calculation-result-preview-errors
                  [errors]="calculationFailureData.errors"
                  [overlayData]="overlayData$ | ngrxPush"
                  [(expanded)]="errorsExpanded"
                ></ea-calculation-result-preview-errors>
              </div>
            }
            @for (item of overlayData; track item) {
              <div
                [class.hidden]="
                  (calculationFailureData?.errors?.length > 0 &&
                    !item.title.includes('emissions')) ||
                  (calculationFailureData?.errors?.length && errorsExpanded)
                "
              >
                @if (!item.title.includes('overrolling')) {
                  <div class="snap-center my-1 md:my-0 min-w-max">
                    <ea-calculation-result-preview-item
                      styleClass="w-[70vw] sm:w-[50vw]"
                      [item]="item"
                    >
                      @if (item.title.includes('emissions')) {
                        <ea-calculation-result-preview-emissions-tooltip>
                        </ea-calculation-result-preview-emissions-tooltip>
                      }
                    </ea-calculation-result-preview-item>
                  </div>
                } @else {
                  <div class="snap-center my-1 md:my-0 min-w-[220px]">
                    <ea-overrolling-frequencies-preview-item
                      [item]="item"
                      styleClass="w-[70vw] sm:w-[50vw]"
                    />
                  </div>
                }
                <mat-divider
                  class="hidden md:block"
                  [vertical]="true"
                ></mat-divider>
              </div>
            }
          </div>
        }
      </ng-container>
      <div
        class="result-button w-full md:w-auto flex flex-shrink-0 flex-grow md:flex-grow-0 md:justify-end items-center md:ml-auto md:mr-0"
      >
        <button
          mat-raised-button
          color="primary"
          class="grow md:mx-0 mx-4"
          [class.hidden]="
            calculationFailureData?.errors?.length > 0 && errorsExpanded
          "
          [disabled]="
            (isCalculationResultReportAvailable$ | ngrxPush) === false
          "
          (click)="showReport()"
        >
          <div class="flex items-center justify-center flex-grow">
            {{ t('showReport') }}
            @if (isAnyServiceLoading$ | ngrxPush) {
              <mat-spinner class="ml-1" diameter="16"></mat-spinner>
            }
          </div>
        </button>
      </div>
    </div>
  </div>
</ng-container>
