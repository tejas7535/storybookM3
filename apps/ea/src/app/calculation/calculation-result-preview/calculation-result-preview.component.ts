import { Dialog } from '@angular/cdk/dialog';
import { CommonModule } from '@angular/common';
import { Component, Input } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatDividerModule } from '@angular/material/divider';
import { MatIconModule } from '@angular/material/icon';
import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';

import { Capacitor } from '@capacitor/core';
import { TrackingService } from '@ea/core/services/tracking-service/tracking.service';
import { CalculationParametersFacade, SettingsFacade } from '@ea/core/store';
import { CalculationResultFacade } from '@ea/core/store/facades/calculation-result/calculation-result.facade';
import { LetDirective, PushPipe } from '@ngrx/component';

import { SharedTranslocoModule } from '@schaeffler/transloco';

import { CalculationResultPreviewErrorsComponent } from '../calculation-result-preview-errors/calculation-result-preview-errors.component';
import { CalculationResultPreviewItemComponent } from '../calculation-result-preview-item/calculation-result-preview-item.component';
import { CalculationResultReportComponent } from '../calculation-result-report/calculation-result-report.component';
import { OverrollingFrequenciesPreviewItemComponent } from '../overrolling-frequencies-preview-iterm/overrolling-frequencies-preview-item.component';

@Component({
  selector: 'ea-calculation-result-preview',
  templateUrl: './calculation-result-preview.component.html',
  imports: [
    CommonModule,
    PushPipe,
    MatIconModule,
    MatButtonModule,
    MatDividerModule,
    CalculationResultPreviewItemComponent,
    OverrollingFrequenciesPreviewItemComponent,
    SharedTranslocoModule,
    MatProgressSpinnerModule,
    LetDirective,
    CalculationResultPreviewErrorsComponent,
  ],
})
export class CalculationResultPreviewComponent {
  @Input() sticky = true;
  public isMobile = Capacitor.isNativePlatform();
  public overlayData$ =
    this.calculationResultFacade.getCalculationResultPreviewData$;

  public isCalculationResultReportAvailable$ =
    this.calculationResultFacade.isCalculationResultReportAvailable$;
  public isAnyServiceLoading$ =
    this.calculationParametersFacade.isAnyServiceLoading$;
  public isCalculationImpossible$ =
    this.calculationResultFacade.isCalculationImpossible$;
  public isCalculationMissingInput$ =
    this.calculationParametersFacade.isCalculationMissingInput$;
  public loadcaseCount$ = this.calculationParametersFacade.getLoadcaseCount$;
  public selectedLoadcase$ =
    this.calculationParametersFacade.getSelectedLoadcase$;
  public calculationErrors$ =
    this.calculationResultFacade.filteredCalculationReportErrors$;
  public calculationDownstreamErrors$ =
    this.calculationResultFacade.calculationReportDownstreamErrors$;
  public calculationGeneralError$ =
    this.calculationResultFacade.isCalculationGeneralError$;

  public isStandalone$ = this.settingsFacade.isStandalone$;

  constructor(
    private readonly calculationResultFacade: CalculationResultFacade,
    private readonly calculationParametersFacade: CalculationParametersFacade,
    private readonly analyticsService: TrackingService,
    private readonly settingsFacade: SettingsFacade,
    private readonly dialog: Dialog
  ) {}

  showReport() {
    this.analyticsService.logShowReport();
    const panelClass = this.isMobile ? 'mobile-report' : '';
    this.dialog.open(CalculationResultReportComponent, {
      id: 'result-report',
      panelClass,
    });
  }
}
