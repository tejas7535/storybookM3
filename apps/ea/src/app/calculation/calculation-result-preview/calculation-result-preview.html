<ng-container *transloco="let t; read: 'calculation'">
  <div
    class="blur-backdrop"
    [ngClass]="{'scale-y-0': !sticky}"
    *ngIf="isStandalone$ | ngrxPush"
  ></div>
  <div
    class="left-0 flex w-full items-center justify-center box-border bottom-0 md:bottom-9 z-20 transition-all duration-150"
    [ngClass]="{'md:inset-y-auto relative md:fixed max-w-[100vw]': (isStandalone$ |ngrxPush) && !sticky, 'relative h-auto': !isStandalone$ | ngrxPush , 'fixed': isStandalone$ | ngrxPush}"
  >
    <div
      class="relative bar bg-background-dark flex flex-col items-center md:flex-row pb-2 px-2 pt-4 md:py-6 md:px-8 rounded-t-lg min-w-full md:min-w-[800px] md:max-w-[calc(100vw-350px)] min-[1000px]:min-w-[890px] drop-shadow-lg z-50 space-y-1 md:space-y-0 overflow-hidden md:rounded-b-lg overflow-y-visible md:mx-4"
      [ngClass]="{'rounded-b-lg rounded-none drop-shadow-none md:rounded-lg md:drop-shadow-lg': 'sticky', 'mx-auto': !isStandalone$ | ngrxPush}"
    >
      <div class="user-hints flex items-center flex-grow-0 flex-shrink-0">
        <ng-container *ngIf="isCalculationMissingInput$ | ngrxPush">
          <div class="flex max-w-[180px] flex-row items-center justify-center">
            <div class="flex flex-shrink-0">
              <mat-icon color="primary">info_outline</mat-icon>
            </div>
            <div
              class="text-caption text-high-emphasis ml-2 text-center md:text-left"
            >
              {{ t('calculationMissingInput') }}
            </div>
          </div>
          <mat-divider class="hidden md:block" [vertical]="true"></mat-divider>
        </ng-container>

        <!-- Calculation Impossible -->
        <ng-container *ngIf="isCalculationImpossible$ | ngrxPush">
          <div class="flex flex-shrink-0 flex-row items-center">
            <div class="flex flex-shrink-0">
              <mat-icon color="error">warning</mat-icon>
            </div>
            <div
              class="text-caption text-high-emphasis ml-2 flex-shrink-0"
              [style]="{ 'max-width': '220px' }"
            >
              {{ t('calculationImpossible') }}
            </div>
          </div>
          <mat-divider [vertical]="true"></mat-divider>
        </ng-container>
      </div>
      <div
        class="result-data"
        *ngIf="overlayData$ | ngrxPush as overlayData"
        class="flex touch-pan-x snap-x snap-mandatory flex-row gap-2 md:max-w-auto mx-2 overflow-x-auto max-w-full md:ml-0 md:mr-4"
      >
        <ng-container *ngFor="let item of overlayData">
          <ng-container
            *ngIf="
                  !item.title.includes('overrolling');
                  else overrollingFrequenciesPreview
                "
          >
            <div class="snap-center my-1 md:my-0 min-w-max">
              <ea-calculation-result-preview-item
                styleClass="w-[70vw] sm:w-[50vw]"
                [item]="item"
              >
                <ea-calculation-result-preview-emissions-tooltip
                  *ngIf="item.title.includes('emissions')"
                >
                </ea-calculation-result-preview-emissions-tooltip>
              </ea-calculation-result-preview-item>
            </div>
          </ng-container>

          <ng-template #overrollingFrequenciesPreview>
            <div class="snap-center my-1 md:my-0 min-w-[220px]">
              <ea-overrolling-frequencies-preview-item
                [item]="item"
                styleClass="w-[70vw] sm:w-[50vw]"
              />
            </div>
          </ng-template>

          <mat-divider class="hidden md:block" [vertical]="true"></mat-divider>
        </ng-container>
      </div>
      <div
        class="result-button w-full md:w-auto flex flex-shrink-0 flex-grow md:flex-grow-0 md:justify-end items-center md:ml-auto md:mr-0"
      >
        <button
          mat-raised-button
          color="primary"
          class="grow md:mx-0 mx-4"
          [disabled]="
              (isCalculationResultReportAvailable$ | ngrxPush) === false
            "
          (click)="showReport()"
        >
          <div class="flex items-center justify-center flex-grow">
            {{ t('showReport') }}
            <mat-spinner
              *ngIf="isAnyServiceLoading$ | ngrxPush"
              class="ml-1"
              diameter="16"
            />
          </div>
        </button>
      </div>
    </div>
  </div>
</ng-container>
