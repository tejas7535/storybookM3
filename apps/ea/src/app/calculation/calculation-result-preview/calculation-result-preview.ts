import { Dialog } from '@angular/cdk/dialog';
import { CommonModule } from '@angular/common';
import { Component, Input } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatDividerModule } from '@angular/material/divider';
import { MatIconModule } from '@angular/material/icon';
import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';

import { EmbeddedGoogleAnalyticsService } from '@ea/core/services/embedded-google-analytics';
import { CalculationParametersFacade, SettingsFacade } from '@ea/core/store';
import { CalculationResultFacade } from '@ea/core/store/facades/calculation-result/calculation-result.facade';
import { PushPipe } from '@ngrx/component';

import { SharedTranslocoModule } from '@schaeffler/transloco';

import { CalculationResultPreviewItemComponent } from '../calculation-result-preview-item/calculation-result-preview-item.component';
import { CalculationResultReportComponent } from '../calculation-result-report/calculation-result-report.component';
import { OverrollingFrequenciesPreviewItemComponent } from '../overrolling-frequencies-preview-iterm/overrolling-frequencies-preview-item.component';

@Component({
  selector: 'ea-calculation-result-preview',
  templateUrl: './calculation-result-preview.html',
  standalone: true,
  imports: [
    CommonModule,
    PushPipe,
    MatIconModule,
    MatButtonModule,
    MatDividerModule,
    CalculationResultPreviewItemComponent,
    OverrollingFrequenciesPreviewItemComponent,
    SharedTranslocoModule,
    MatProgressSpinnerModule,
  ],
})
export class CalculationResultPreviewComponent {
  @Input() sticky = true;

  public overlayData$ =
    this.calculationResultFacade.getCalculationResultPreviewData$;
  public isCalculationResultReportAvailable$ =
    this.calculationResultFacade.isCalculationResultReportAvailable$;
  public isAnyServiceLoading$ =
    this.calculationParametersFacade.isAnyServiceLoading$;
  public isCalculationImpossible$ =
    this.calculationResultFacade.isCalculationImpossible$;
  public isCalculationMissingInput$ =
    this.calculationParametersFacade.isCalculationMissingInput$;
  public loadcaseCount$ = this.calculationParametersFacade.getLoadcaseCount$;
  public selectedLoadcase$ =
    this.calculationParametersFacade.getSelectedLoadcase$;

  public isStandalone$ = this.settingsFacade.isStandalone$;

  constructor(
    private readonly calculationResultFacade: CalculationResultFacade,
    private readonly calculationParametersFacade: CalculationParametersFacade,
    private readonly analyticsService: EmbeddedGoogleAnalyticsService,
    private readonly settingsFacade: SettingsFacade,
    private readonly dialog: Dialog
  ) {}

  showReport() {
    this.analyticsService.logShowReport();
    this.dialog.open(CalculationResultReportComponent, { id: 'result-report' });
  }
}
