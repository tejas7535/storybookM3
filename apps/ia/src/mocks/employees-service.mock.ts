import { Employee } from '../app/shared/models';

export const ROOT = ({
  employeeId: '123',
  parentEmployeeId: undefined,
  exitDate: undefined,
  totalSubordinates: 0,
  directSubordinates: 0,
  directAttrition: 0,
  totalAttrition: 0,
} as unknown) as Employee;

export const LEVEL_2_EMPLOYEE_A = ({
  employeeId: '456',
  parentEmployeeId: '123',
  exitDate: undefined,
  totalSubordinates: 0,
  directSubordinates: 0,
  directAttrition: 0,
  totalAttrition: 0,
} as unknown) as Employee;

export const LEVEL_2_EMPLOYEE_B = ({
  employeeId: '789',
  parentEmployeeId: '123',
  exitDate: undefined,
  totalSubordinates: 0,
  directSubordinates: 0,
  directAttrition: 0,
  totalAttrition: 0,
} as unknown) as Employee;

export const LEVEL_2_EMPLOYEE_C = ({
  employeeId: '13',
  parentEmployeeId: '123',
  exitDate: '2015-10-10',
  totalSubordinates: 0,
  directSubordinates: 0,
  directAttrition: 0,
  totalAttrition: 0,
  entryDate: '2010-10-10',
  terminationDate: '2015-08-10',
} as unknown) as Employee;

export const LEVEL_3_EMPLOYEE_A = ({
  employeeId: '33',
  parentEmployeeId: '789',
  exitDate: '2015-10-10',
  totalSubordinates: 0,
  directSubordinates: 0,
  directAttrition: 0,
  totalAttrition: 0,
} as unknown) as Employee;

export const EMPLOYEES: Employee[] = [
  ROOT,
  LEVEL_2_EMPLOYEE_A,
  LEVEL_2_EMPLOYEE_B,
  LEVEL_2_EMPLOYEE_C,
  LEVEL_3_EMPLOYEE_A,
];

const map = new Map<string, Employee[]>();
map.set('-1', [ROOT]);
map.set('123', [LEVEL_2_EMPLOYEE_A, LEVEL_2_EMPLOYEE_B, LEVEL_2_EMPLOYEE_C]);
map.set('789', [LEVEL_3_EMPLOYEE_A]);

export const EMPLOYEE_MAP = map;
