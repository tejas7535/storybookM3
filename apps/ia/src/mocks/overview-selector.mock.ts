import {
  Employee,
  EmployeeAttritionMeta,
  LeavingType,
} from '../app/shared/models';

export const EMPLOYEES_FOR_SELECTORS: Employee[] = [
  ({
    employeeId: '123',
    parentEmployeeId: undefined,
    orgUnit: 'Schaeffler_IT',
    directSubordinates: 3,
    totalSubordinates: 4,
    directAttrition: 1,
    totalAttrition: 2,
    exitDate: undefined,
    directLeafChildren: [],
  } as unknown) as Employee,
  ({
    employeeId: '456',
    parentEmployeeId: '123',
    orgUnit: 'Schaeffler_IT_1',
    directSubordinates: 0,
    totalSubordinates: 0,
    directAttrition: 0,
    totalAttrition: 0,
    exitDate: new Date('2020-02-28'),
    reasonForLeaving: LeavingType.FORCED,
    directLeafChildren: [],
  } as unknown) as Employee,
  ({
    employeeId: '678',
    parentEmployeeId: '123',
    orgUnit: 'Schaeffler_IT_1',
    directSubordinates: 1,
    totalSubordinates: 1,
    directAttrition: 1,
    totalAttrition: 1,
    exitDate: undefined,
    directLeafChildren: [],
  } as unknown) as Employee,
  ({
    employeeId: '999',
    parentEmployeeId: '123',
    orgUnit: 'Schaeffler_IT_1',
    directSubordinates: 0,
    totalSubordinates: 0,
    directAttrition: 0,
    totalAttrition: 0,
    exitDate: undefined,
    entryDate: new Date('2020-06-06'),
    directLeafChildren: [],
  } as unknown) as Employee,
  ({
    employeeId: '91011',
    parentEmployeeId: '678',
    orgUnit: 'Schaeffler_IT_1_2',
    directSubordinates: 0,
    totalSubordinates: 0,
    directAttrition: 0,
    totalAttrition: 0,
    exitDate: new Date('2020-05-05'),
    reasonForLeaving: LeavingType.UNFORCED,
    directLeafChildren: [],
  } as unknown) as Employee,
];

export const EXPECTED_FILTERED_EMPLOYEES: Employee[] = [
  ({
    employeeId: '123',
    exitDate: undefined,
    parentEmployeeId: undefined,
    orgUnit: 'Schaeffler_IT',
    directSubordinates: 2,
    totalSubordinates: 2,
    directAttrition: 1,
    totalAttrition: 2,
    directLeafChildren: [
      ({
        employeeId: '999',
        parentEmployeeId: '123',
        orgUnit: 'Schaeffler_IT_1',
        directSubordinates: 0,
        totalSubordinates: 0,
        directAttrition: 0,
        totalAttrition: 0,
        entryDate: new Date('2020-06-06'),
        exitDate: undefined,
        directLeafChildren: [],
        attritionMeta: new EmployeeAttritionMeta(
          'Schaeffler_IT_1',
          0,
          0,
          0,
          1,
          1,
          0,
          1,
          0
        ),
      } as unknown) as Employee,
    ],
    attritionMeta: new EmployeeAttritionMeta(
      'Schaeffler_IT',
      45.28,
      2,
      0,
      1,
      1,
      0,
      1,
      0
    ),
  } as unknown) as Employee,
  ({
    employeeId: '678',
    parentEmployeeId: '123',
    orgUnit: 'Schaeffler_IT_1',
    directSubordinates: 0,
    totalSubordinates: 0,
    directAttrition: 1,
    totalAttrition: 1,
    directLeafChildren: [],
    exitDate: undefined,
    attritionMeta: new EmployeeAttritionMeta(
      'Schaeffler_IT_1',
      29.27,
      1,
      0,
      1,
      1,
      0,
      1,
      0
    ),
  } as unknown) as Employee,
];
