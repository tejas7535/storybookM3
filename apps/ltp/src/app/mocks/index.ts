export * from 'mock.constants';
export * from 'mock.service';
