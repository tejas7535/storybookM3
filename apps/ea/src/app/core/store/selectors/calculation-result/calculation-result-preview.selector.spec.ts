import { APP_STATE_MOCK } from '@ea/testing/mocks';

import { getCalculationResultPreviewData } from './calculation-result-preview.selector';

describe('Calculation Result Selector', () => {
  const mockState = {
    ...APP_STATE_MOCK,
  };

  describe('getCalculationResultPreviewData', () => {
    it('should return the result preview data', () => {
      expect(getCalculationResultPreviewData(mockState)).toMatchSnapshot();
    });
  });
});
