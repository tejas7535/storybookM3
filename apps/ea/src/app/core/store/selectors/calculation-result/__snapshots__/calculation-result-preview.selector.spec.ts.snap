// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculation Result Selector co2Upstream should return the upstream co2 result 1`] = `
Object {
  "calculationError": undefined,
  "isLoading": false,
  "unit": "kg",
  "value": 2,
}
`;

exports[`Calculation Result Selector co2Upstream should return undefined if no upstream co2 result is available 1`] = `
Object {
  "calculationError": undefined,
  "isLoading": undefined,
  "unit": undefined,
  "value": undefined,
}
`;

exports[`Calculation Result Selector getCalculationResultPreviewData should return the result preview data 1`] = `
Array [
  Object {
    "icon": "compress",
    "title": "frictionalPowerloss",
    "values": Array [
      Object {
        "calculationError": undefined,
        "isLoading": false,
        "title": "frictionalPowerlossSubtitle",
        "unit": undefined,
        "value": undefined,
      },
    ],
  },
  Object {
    "svgIcon": "co2",
    "title": "emissions",
    "values": Array [
      Object {
        "calculationError": undefined,
        "isLoading": false,
        "title": "production",
        "unit": "kg",
        "value": 2,
      },
    ],
  },
]
`;
